### ISSUE TITLE
CtExecutableReference cannot use CtTypeParameterReference. Use boundingType of CtTypeParameterReference instead

### ISSUE THREAD

**Issue by surli** [INRIA/spoon#2721]
For the record our CI using Spoon against Guava is in bad shape since the beginning of the month. See: https://ci.inria.fr/sos/job/Guava/

Last error we got is the following:

```
[ERROR] Failed to execute goal fr.inria.gforge.spoon:spoon-maven-plugin:2.4:generate (default) on project guava: Exception during the spoonify of the target project. spoon.SpoonException: CtExecutableReference cannot use CtTypeParameterReference. Use boundingType of CtTypeParameterReference instead. -> [Help 1]
```

**Comment by pvojtechovsky** [comment]
Is anybody able to provide full stacktrace of this problem? I guess it will be much easier to fix it then. Actually it needs some effort to understand and create the same Guava testing environment and then run it to get full stacktrace.

**Comment by monperrus** [comment]
> Actually it needs some effort to understand and create the same Guava testing environment and then run it to get full stacktrace.

good point! see #2724

**Comment by surli** [comment]
Here's the stacktrace:

```
[ERROR] Failed to execute goal fr.inria.gforge.spoon:spoon-maven-plugin:2.4:generate (default) on project guava: Exception during the spoonify of the target project. spoon.SpoonException: CtExecutableReference cannot use CtTypeParameterReference. Use boundingType of CtTypeParameterReference instead. -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal fr.inria.gforge.spoon:spoon-maven-plugin:2.4:generate (default) on project guava: Exception during the spoonify of the target project.
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Exception during the spoonify of the target project.
	at fr.inria.gforge.spoon.Spoon.execute(Spoon.java:159)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 20 more
Caused by: org.apache.maven.plugin.MojoExecutionException: Exception during the spoonify of the target project.
	at fr.inria.gforge.spoon.metrics.PerformanceDecorator.execute(PerformanceDecorator.java:24)
	at fr.inria.gforge.spoon.Spoon.execute(Spoon.java:154)
	... 22 more
Caused by: spoon.SpoonException: spoon.SpoonException: CtExecutableReference cannot use CtTypeParameterReference. Use boundingType of CtTypeParameterReference instead.
	at spoon.Launcher.prettyprint(Launcher.java:786)
	at spoon.Launcher.run(Launcher.java:722)
	at fr.inria.gforge.spoon.metrics.PerformanceDecorator.execute(PerformanceDecorator.java:22)
	... 23 more
Caused by: spoon.SpoonException: CtExecutableReference cannot use CtTypeParameterReference. Use boundingType of CtTypeParameterReference instead.
	at spoon.support.reflect.reference.CtExecutableReferenceImpl.checkMethodParameterTypeRef(CtExecutableReferenceImpl.java:193)
	at spoon.support.reflect.reference.CtExecutableReferenceImpl.addParameter(CtExecutableReferenceImpl.java:185)
	at spoon.support.reflect.reference.CtExecutableReferenceImpl.setParameters(CtExecutableReferenceImpl.java:176)
	at spoon.reflect.factory.ExecutableFactory.createReference(ExecutableFactory.java:199)
	at spoon.reflect.factory.ExecutableFactory.createReference(ExecutableFactory.java:175)
	at spoon.reflect.factory.ExecutableFactory.createReferenceInternal(ExecutableFactory.java:113)
	at spoon.reflect.factory.ExecutableFactory.createReference(ExecutableFactory.java:99)
	at spoon.support.reflect.declaration.CtExecutableImpl.getReference(CtExecutableImpl.java:191)
	at spoon.reflect.visitor.ImportScannerImpl.addClassImport(ImportScannerImpl.java:398)
	at spoon.reflect.visitor.MinimalImportScanner.addClassImport(MinimalImportScanner.java:49)
	at spoon.reflect.visitor.ImportScannerImpl.visitCtTypeReference(ImportScannerImpl.java:138)
	at spoon.support.reflect.reference.CtTypeReferenceImpl.accept(CtTypeReferenceImpl.java:77)
	at spoon.reflect.visitor.ImportScannerImpl.scan(ImportScannerImpl.java:148)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:173)
	at spoon.reflect.visitor.CtScanner.visitCtMethod(CtScanner.java:556)
	at spoon.support.reflect.declaration.CtMethodImpl.accept(CtMethodImpl.java:67)
	at spoon.reflect.visitor.ImportScannerImpl.scan(ImportScannerImpl.java:148)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:173)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:147)
	at spoon.reflect.visitor.CtScanner.visitCtClass(CtScanner.java:338)
	at spoon.reflect.visitor.ImportScannerImpl.visitCtClass(ImportScannerImpl.java:258)
	at spoon.support.reflect.declaration.CtClassImpl.accept(CtClassImpl.java:67)
	at spoon.reflect.visitor.ImportScannerImpl.scan(ImportScannerImpl.java:148)
	at spoon.reflect.visitor.ImportScannerImpl.computeImports(ImportScannerImpl.java:311)
	at spoon.reflect.visitor.DefaultJavaPrettyPrinter.computeImports(DefaultJavaPrettyPrinter.java:326)
	at spoon.reflect.visitor.DefaultJavaPrettyPrinter.calculate(DefaultJavaPrettyPrinter.java:1982)
	at spoon.support.JavaOutputProcessor.createJavaFile(JavaOutputProcessor.java:120)
	at spoon.support.JavaOutputProcessor.process(JavaOutputProcessor.java:156)
	at spoon.support.JavaOutputProcessor.process(JavaOutputProcessor.java:46)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:74)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:173)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:147)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:649)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:97)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:181)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:77)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:173)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:147)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:648)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:97)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:181)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:77)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:173)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:147)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:648)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:97)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:181)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:77)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:173)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:147)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:648)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:97)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:181)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:77)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:173)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:147)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:648)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:97)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:181)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:77)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:173)
	at spoon.reflect.visitor.CtScanner.visitCtModule(CtScanner.java:923)
	at spoon.reflect.factory.ModuleFactory$CtUnnamedModule.accept(ModuleFactory.java:105)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:181)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:77)
	at spoon.support.QueueProcessingManager.process(QueueProcessingManager.java:129)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.generateProcessedSourceFilesUsingTypes(JDTBasedSpoonCompiler.java:515)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.generateProcessedSourceFiles(JDTBasedSpoonCompiler.java:210)
	at spoon.Launcher.prettyprint(Launcher.java:784)
	... 25 more
```

**Comment by pvojtechovsky** [comment]
fixed by #2725
